File: src/apps/scala/slate-sampleapps/SampleApp.Core/src/main/scala/sampleapp/core/services/MovieApi.scala
/**
  * <slate_header>
  * author: Kishore Reddy
  * url: https://github.com/kishorereddy/scala-slate
  * copyright: 2016 Kishore Reddy
  * license: https://github.com/kishorereddy/scala-slate/blob/master/LICENSE.md
  * desc: a scala micro-framework
  * usage: Please refer to license on github for more info.
  * </slate_header>
  */
package sampleapp.core.services

import sampleapp.core.models.Movie
import slate.core.apis.{ApiAction, Api}
import slate.core.common.svcs.ApiEntityWithSupport
import scala.reflect.runtime.universe.{typeOf}

@Api(area = "sampleapp", name = "movies", desc = "api for users", roles= "*", auth = "app-roles", verb = "post", protocol = "*")
class MovieApi extends ApiEntityWithSupport[Movie, MovieService] {

  override def init():Unit =
  {
    val svc = context.ent.getService(typeOf[Movie]).asInstanceOf[MovieService]
    initContext(svc)
  }
}

File: src/apps/kotlin/slatekit-sampleapps/sampleapp-core/src/main/kotlin/slatekit/sampleapp/core/services/MovieApi.kt
/**
  * <slate_header>
  * author: Kishore Reddy
  * url: https://github.com/kishorereddy/scala-slate
  * copyright: 2016 Kishore Reddy
  * license: https://github.com/kishorereddy/scala-slate/blob/master/LICENSE.md
  * desc: a scala micro-framework
  * usage: Please refer to license on github for more info.
  * </slate_header>
  */
package slatekit.sampleapp.core.services

import slatekit.apis.Api
import slatekit.apis.ApiAction
import slatekit.common.DateTime
import slatekit.integration.common.ApiBaseEntity
import slatekit.integration.common.AppEntContext
import slatekit.sampleapp.core.models.Movie


@Api(area = "app", name = "movies", desc = "api for users", roles= "*", auth = "app-roles", verb = "post", protocol = "*")
class MovieApi( context: AppEntContext) : ApiBaseEntity<Movie, MovieService>(context, Movie::class)
{
    @ApiAction(roles = "", verb = "@parent", protocol = "*")
    fun createSample(title:String, category:String, playing:Boolean, cost:Int, rating:Double, released: DateTime):Long {
        return service.create(Movie(title    = title,
              category = category,
              playing  = playing,
              cost     = cost,
              rating   = rating,
              released = released
        ))
    }
}